### PR TITLE
Add userEnvProbe, indicate what works in Codespaces

### PR DESCRIPTION
**Pull request by Chuxel**
I took a crack at trying to indicate which devcontainer.json properties currently work in Codespaces using a caution symbol. While adding a column would be ideal, we're cramped for horizonal space.

### CODE REVIEW

**Comment by Chuxel** [comment]
@gregvanl Yeah that sounds reasonable.  Will update.

**Comment by Chuxel** [comment]
@gregvanl Updated! Found a few other things while going back through that I also updated.

**Review by gregvanl — APPROVED** [review]

**Review by Zenix27 — APPROVED** [review]

File: docs/remote/devcontainerjson-reference.md [hunk 1]
```diff
@@ -29,29 +31,30 @@ See [Set up a folder to run in a container](/docs/remote/create-dev-container.md
...
+| `workspaceMount` | string | Overrides the default local mount point for the workspace when the container is created. Supports the same values as the [Docker CLI `--mount` flag](https://docs.docker.com/engine/reference/commandline/run/#add-bind-mounts-or-volumes-using-the---mount-flag). Primarily useful for [configuring remote containers](/docs/remote/containers-advanced.md#developing-inside-a-container-on-a-remote-docker-host) or [improving disk performance](/docs/remote/containers-advanced.md#improving-container-disk-performance). Environment and [pre-defined variables](#variables-in-devcontainerjson) may be referenced in the value. For example: <br />`"workspaceMount": "source=${localWorkspaceFolder}/sub-folder,target=/workspace,type=bind,consistency=cached"`<br /><br />⚠️ Not yet supported in Codespaces or when using Open Repository in Container. |
```
Comment (bamurtaugh) [inline docs/remote/devcontainerjson-reference.md:35]: Did we update `Open Repository in Container` to `Clone Repository in Container Volume`?

File: docs/remote/devcontainerjson-reference.md [hunk 2]
```diff
@@ -64,10 +67,10 @@ Variables can be referenced in certain string values in `devcontainer.json` in t
...
 | `${containerEnv:VARIABLE_NAME}` | `remoteEnv` | Value of an existing environment variable inside the container once it is up and running (in this case, called `VARIABLE_NAME`). For example:<br /> `"remoteEnv": { "PATH": "${containerEnv:PATH}:/some/other/path" }` |
-| `${localWorkspaceFolder}` | Any | Path of the local folder that was opened in VS Code (that contains `.devcontainer/devcontainer.json`). |
-| `${containerWorkspaceFolder}` | Any | The path that the workspaces files can be found in the container. |
-| `${localWorkspaceFolderBasename}` | Any | Name of the local folder that was opened in VS Code (that contains `.devcontainer/devcontainer.json`). |
-| `${containerWorkspaceFolderBasename}` | Any | Name of the folder where the workspace files can be found in the container. |
+| `${localWorkspaceFolder}`  | Any | Path of the local folder that was opened in VS Code (that contains `.devcontainer/devcontainer.json`).<br /><br />⚠️ Not yet supported in Codespaces or when using Open Repository in Container. |
```
Comment (bamurtaugh) [inline docs/remote/devcontainerjson-reference.md:70]: Update to `Clone Repository in Container Volume`?

File: docs/remote/devcontainerjson-reference.md [hunk 3]
```diff
@@ -64,10 +67,10 @@ Variables can be referenced in certain string values in `devcontainer.json` in t
...
+| `${containerWorkspaceFolder}` | Any | The path that the workspaces files can be found in the container.<br /><br />⚠️ Not supported in Codespaces yet. |
+| `${localWorkspaceFolderBasename}` | Any | Name of the local folder that was opened in VS Code (that contains `.devcontainer/devcontainer.json`).<br /><br />⚠️ Not yet supported in Codespaces or when using Open Repository in Container. |
```
Comment (bamurtaugh) [inline docs/remote/devcontainerjson-reference.md:72]: Update to `Clone Repository in Container Volume`?

**Review by bamurtaugh — APPROVED** [review]
I believe we update one of the command names for Remote - Containers so I made a couple of suggestions there- I think otherwise looks great to me!

**Comment by Chuxel** [comment]
> Did we update `Open Repository in Container` to `Clone Repository in Container Volume`? 

@bamurtaugh Good catch! Updated.

**Pull request merged**